### PR TITLE
Fix etcdWatcher test

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix https://github.com/kubernetes/kubernetes/issues/18929

@timothysc

### CODE REVIEW

File: pkg/storage/etcd/etcd_watcher_test.go [hunk 1]
```diff
@@ -266,8 +266,15 @@ func TestWatch(t *testing.T) {
 
 	watching.Stop()
 
-	if _, open := <-watching.ResultChan(); open {
-		t.Errorf("An injected error did not cause a graceful shutdown")
+	// There is a race in etcdWatcher so that after calling Stop() one of
+	// two can happen:
+	// - ResultChan() may be close (triggered by closing userStop channel)
```
Comment (gmarek) [inline pkg/storage/etcd/etcd_watcher_test.go]: s/close/closed/

File: pkg/storage/etcd/etcd_watcher_test.go [hunk 2]
```diff
@@ -266,8 +266,15 @@ func TestWatch(t *testing.T) {
 
 	watching.Stop()
 
-	if _, open := <-watching.ResultChan(); open {
-		t.Errorf("An injected error did not cause a graceful shutdown")
+	// There is a race in etcdWatcher so that after calling Stop() one of
+	// two can happen:
```
Comment (gmarek) [inline pkg/storage/etcd/etcd_watcher_test.go]: s/two can/two things can/

**Comment by gmarek** [comment]
two nits, LGTM otherwise.

File: pkg/storage/etcd/etcd_watcher_test.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/storage/etcd/etcd_watcher_test.go]: done

File: pkg/storage/etcd/etcd_watcher_test.go (hunk 2 above)
Comment (wojtek-t) [inline pkg/storage/etcd/etcd_watcher_test.go]: done

**Comment by wojtek-t** [comment]
@gmarek - PTAL

**Comment by wojtek-t** [comment]
> git -c core.askpass=true fetch --tags --progress https://github.com/kubernetes/kubernetes +refs/pull/*:refs/remotes/origin/pr/*
ERROR: Error fetching remote repo 'origin'

@k8s-bot unit test this please

**Comment by gmarek** [comment]
We get this error quite often now:/
LGTM

**Pull request merged**